spot_ros2_control/examples/src/noarm_squat.cpp

```diff
@@ -0,0 +1,143 @@
+    }
+    if (squat_joint_angles_.size() != njoints_) {
+      throw std::logic_error("Squat joint angles is the wrong size!");
+    }
+
+    points_per_motion_ = static_cast<int>(command_rate * seconds_per_motion);
+    command_.data = std::vector<double>(njoints_, 0.0);
+
+    command_pub_ = create_publisher<std_msgs::msg::Float64MultiArray>("/forward_position_controller/commands", 10);
+    joint_states_sub_ = create_subscription<sensor_msgs::msg::JointState>(
+        "joint_states", 10, std::bind(&NoarmSquat::joint_states_callback, this, std::placeholders::_1));
+
+    const auto timer_period =
+        std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::duration<double>(1. / command_rate));
+    timer_ = create_wall_timer(timer_period, std::bind(&NoarmSquat::timer_callback, this));
+  }
+
+ private:
+  // For storing joint angles
+  std::vector<double> stand_joint_angles_;
+  std::vector<double> squat_joint_angles_;
+  std::vector<double> init_joint_angles_;
+  // Command we send to the robot
+  std_msgs::msg::Float64MultiArray command_;
+  // Parameters and info about state
+  SquatState squat_state_;
+  int points_per_motion_;
+  bool initialized_;
+  int count_;
+  size_t njoints_;
+  // RCLCPP publishers, subscribers, timers
+  rclcpp::TimerBase::SharedPtr timer_;
+  rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_states_sub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr command_pub_;
+
+  /// @brief Callback for receiving joint states messages used to store the nominal joint angles of the robot
+  /// @param msg ROS message containing joint states
+  void joint_states_callback(const sensor_msgs::msg::JointState& msg) {
+    if (!initialized_) {
+      RCLCPP_INFO_STREAM(get_logger(), "Received starting joint states");
+      init_joint_angles_ = msg.position;
+      initialized_ = true;
+    }
+  }
+
+  /// @brief We start in INITIALIZING (going from initial joint angles to squatting joint angles).
+  /// Then we just switch indefinitely between SQUATTING and SITTING.
+  void state_transition() {
+    switch (squat_state_) {
+      case SquatState::INITIALIZING:
+        squat_state_ = SquatState::SQUATTING;
+        break;
+      case SquatState::SQUATTING:
+        squat_state_ = SquatState::STANDING;
+        break;
+      case SquatState::STANDING:
+        squat_state_ = SquatState::SQUATTING;
+        break;
+    }
+  }
+
+  /// @brief Fills in the command to send to the robot
+  /// @param baseline Vector of joint angles to start at
+  /// @param goal Vector of joint angles to finish at
+  /// @param percentage Percentage of the motion (from 0-1) we are at. 0 corresponds to being at baseline
+  /// and 1 corresponds to being at goal, anything in between is calculated as a linear interpolation between
+  /// the two.
+  void populate_command(std::vector<double>& baseline, std::vector<double>& goal, double percentage) {
+    for (size_t i = 0; i < njoints_; i++) {
```

Comment:
@khughes-bdai nit: as a rule of thumb, if you don't need the previous value, use the pre-increment operator:
```suggestion
    for (size_t i = 0; i < njoints_; ++i) {
```
Same elsewhere. It's also good practice, even if in this specific case the compiler will most likely make them equivalent.